server/plugin/command.go

```diff
@@ -609,14 +610,22 @@ func (p *Plugin) handleMe(_ *plugin.Context, _ *model.CommandArgs, _ []string, u
 	return text
 }
 
-func (p *Plugin) handleHelp(_ *plugin.Context, _ *model.CommandArgs, _ []string, _ *GitHubUserInfo) string {
+func (p *Plugin) handleHelp(_ *plugin.Context, args *model.CommandArgs, _ []string, _ *GitHubUserInfo) string {
+	l := p.b.GetUserLocalizer(args.UserId)
 	message, err := renderTemplate("helpText", p.getConfiguration())
```

Comment:
Curious how we can use the translations in the templates?

Reply:
yeah @mickmister I was trying to find the same thing.

Reply:
Can we expose functions to call from within the template?

Reply:
I think this is something we should try and solve before merging the PR. Any question marks we have on how to do things like this should ideally be resolved before contributors go and work on the same sort of things

Reply:
We can implement this method:

```go
func (p *Plugin) localize(id, other string) string {
	return p.b.LocalizeWithConfig(l, &i18n.LocalizeConfig{
		DefaultMessage: &i18n.Message{
			ID:    id,
			Other: other,
		},
	})
}
```

---

Then change

https://github.com/mattermost/mattermost-plugin-github/blob/e958064b9d26bd23cee76d385e721c7b4e654fee/server/plugin/template.go#L38

to

```go
func (p *Plugin) initTemplates() {
```

---

and add a block here

https://github.com/mattermost/mattermost-plugin-github/blob/e958064b9d26bd23cee76d385e721c7b4e654fee/server/plugin/template.go#L108-L116

to

```go
funcMap["i18n"] = p.localize
```

---

Then we should be able to use the `i18n` function in the templates

Reply:
Hey @mickmister  Thanks for the approach above, I tried a bunch of things as stated above but faced several issues.
First of all, there is a huge lack of documentation and resources on the internet regarding Go's text/template and i was not able to find anything which connects the translation and template together. If you have some resources, please share those i would love to deep dive into those. 

First of all, When using higher-order functions inside the template directly, it prints the function returned by the parent function instead of trying to run the child function returned with the arguments passed(similar code here [dummy code).](https://go.dev/play/p/QSNIDmzwpg8l) I also tried storing the child function in a variable and calling it with the arguments, but that didn't work either. (You can have a link here [goPlayground)](https://go.dev/play/p/D4tr39bUJHl)

Secondly, the approach of making the initTemplate a method of the Plugin also didn't work because, during the initialization of the template, the Plugin is null.

Thirdly, We should not use any word in the template but use it just to pass the values to the function and inside the function add the words according to the localization

Reply:
> Secondly, the approach of making the initTemplate a method of the Plugin also didn't work because, during the initialization of the template, the Plugin is null.

@Kshitij-Katiyar We should be able to call `initTemplate` during `OnActivate` to make that work

I'm not sure about the template functions behavior